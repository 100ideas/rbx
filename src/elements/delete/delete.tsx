import { cx } from "emotion";
import React from "react";

import { forwardRefAs } from "@/exotic";
import { ModifierProps } from "@/modifiers";
import { tuple } from "@/utils";

export const DELETE_SIZES = tuple("small", "medium", "large", "auto");
export type DeleteSizes = (typeof DELETE_SIZES)[number];

export type DeleteModifierProps = Partial<{
  size: DeleteSizes;
}>;

export type DeleteProps = ModifierProps & DeleteModifierProps;

export const Delete = forwardRefAs<DeleteProps, "a">((props, ref) => {
  const { as, size, ...rest } = props;
  rest.className = cx("delete", rest.className, {
    [`is-${size}`]: size !== "auto",
  });
  return React.createElement(as!, { ref, ...rest });
}, "a");

Delete.defaultProps = Object.assign({ size: "auto" }, Delete.defaultProps);
