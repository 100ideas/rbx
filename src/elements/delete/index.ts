import "./delete.module.sass";
export { Delete } from "./delete";
