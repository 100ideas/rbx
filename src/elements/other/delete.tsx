import { cx } from "emotion";
import React from "react";

import { forwardRefAs } from "@/generic";
import { ModifierProps } from "@/modifiers";
import { tuple } from "@/utils";

export const DELETE_SIZES = tuple("small", "medium", "large");
export type DeleteSizes = (typeof DELETE_SIZES)[number];

export type DeleteModifierProps = Partial<{
  size: DeleteSizes;
}>;

export type DeleteProps = ModifierProps & DeleteModifierProps;

export const Delete = forwardRefAs<DeleteProps, "a">((props, ref) => {
  const { as, size, ...rest } = props;
  rest.className = cx("delete", rest.className, {
    [`is-${size}`]: size,
  });
  return React.createElement(as!, { ref, ...rest });
}, "a");
