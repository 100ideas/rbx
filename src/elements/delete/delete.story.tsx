import { select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import React from "react";

import { Delete } from "@/elements";

const makeSizeSelector = () =>
  select(
    "Size",
    {
      auto: "auto",
      small: "small",
      medium: "medium", // tslint:disable-line:object-literal-sort-keys
      large: "large",
    },
    "auto",
  );

storiesOf("Elements/Delete", module).add("Default", () => (
  <div>
    <Delete size={makeSizeSelector()} />
  </div>
));
