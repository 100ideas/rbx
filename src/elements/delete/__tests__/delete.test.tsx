import React from "react";
import renderer from "react-test-renderer";

import { Delete, DELETE_SIZES } from "../delete";

describe("Delete component", () => {
  it("should Exist", () => {
    expect(Delete).toMatchSnapshot();
  });

  it("should have delete classname", () => {
    const component = renderer.create(<Delete />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  DELETE_SIZES.map(size => {
    it(`should be size ${size}`, () => {
      const component = renderer.create(<Delete size={size} />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
