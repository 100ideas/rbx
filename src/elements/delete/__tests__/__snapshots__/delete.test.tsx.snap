// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete component should Exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "as": "a",
    "size": "auto",
  },
  "render": [Function],
}
`;

exports[`Delete component should be size auto 1`] = `
<a
  className="delete"
/>
`;

exports[`Delete component should be size large 1`] = `
<a
  className="delete is-large"
/>
`;

exports[`Delete component should be size medium 1`] = `
<a
  className="delete is-medium"
/>
`;

exports[`Delete component should be size small 1`] = `
<a
  className="delete is-small"
/>
`;

exports[`Delete component should have delete classname 1`] = `
<a
  className="delete"
/>
`;
